CMakeLists.txt

```diff
@@ -1830,7 +1830,13 @@ MYSQL_CHECK_LDAP()
 MYSQL_CHECK_LDAP_DLLS()
 
 # Percona LDAP Simple / LDAP SASL authentication plugins
-OPTION(WITH_PERCONA_AUTHENTICATION_LDAP "Build with Percona LDAP Simple / LDAP SASL authentication plugins" ON)
+# Problem with  MacOS system libraries and complains for compilation error ldap_set_urllist_proc
+IF(APPLE)
+  OPTION(WITH_PERCONA_AUTHENTICATION_LDAP OFF)
+ELSE()
+  OPTION(WITH_PERCONA_AUTHENTICATION_LDAP ON)
+ENDIF()
+OPTION(WITH_PERCONA_AUTHENTICATION_LDAP "Build with Percona LDAP Simple / LDAP SASL authentication plugins")
```

Comment:
@rahulmalik87 To be honest I did not find anywhere in the CMake docs that this option redeclaration (first with value and then with name) is a valid construct.

Reply:
It's somewhat documented in backward compatibility pages. Essentially OPTION creates a cached variable, and will not override existing variables.

In this sense, redeclaration is not an error, but the second declaration won't override the already existing docstring. With this change, that means the docstring will be empty.

Reply:
I checked the docstring in CMakeCache.txt. It was not empty.

`//Build with Percona LDAP Simple / LDAP SASL authentication plugins
WITH_PERCONA_AUTHENTICATION_LDAP:BOOL=OFF`

Reply:
OK, fine with me if it works

Reply:
@rahulmalik87 I rechecked that, you are right. But that is not what the documentation says: "If <variable> is already set as a normal or cache variable, then the command does nothing" Not sure if this is a cmake bug or a documentation bug, as changing the text is not nothing.

Reply:
thanks Guys, Fixed. now using WITH_PERCONA_AUTHENTICATION_LDAP_DEFAULT